File: Kun/leetcode/21-Merge-Two-Sorted-Lists/21-Merge-Two-Sorted-Lists.py
import ListNode

class Solution:
    def mergeTwoLists(self, l1: ListNode, l2: ListNode) -> ListNode:
        pre = dummyHead = ListNode(next = l1)
        while l1 or l2:
            if not l2 or l1 and l1.val < l2.val:
                pre.next = l1
                pre = l1
                l1 = l1.next
            else:
                pre.next = l2
                pre  = l2
                l2 = l2.next
        return dummyHead.next

File: Kun/leetcode/21-Merge-Two-Sorted-Lists/21-Merge-Two-Sorted-Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode dummyHead = new ListNode();
        dummyHead.next = l1;
        ListNode pre = dummyHead;
        while(l1 != null || l2 != null){
            if(l2 == null || l1 != null && l1.val < l2.val){
                pre.next = l1;
                pre = l1;
                l1 = l1.next;
            } else {
                pre.next = l2;
                pre = l2;
                l2 = l2.next;
            }
        }
        return dummyHead.next;
    }
}